tabular/src/autogluon/tabular/task/tabular_prediction/predictor.py

```diff
@@ -816,6 +817,120 @@ def fit_weighted_ensemble(self, base_models: list = None, name_suffix='_custom',
 
         return models
 
+    # TODO: v0.1 This can cause very strange issues related to index mismatches with original training data on extreme edge cases (multiclass where autogluon duplicated rows in data due to rare classes and eval_metric was log_loss)
+    #  This is a very rare case but needs to be fixed by v0.1 release
+    #  It might be good enough to do an inner join on training data, but it needs to be tested
+    def get_oof_pred(self, model: str = None, transformed=False) -> pd.Series:
+        """
+        Note: This is advanced functionality not intended for normal usage.
+
+        Returns the out-of-fold (OOF) training predictions of a model.
+        OOF predictions are unbiased estimates of model predictions on the training data.
+
+        Warning: This method will raise an exception if called on a model that is not a bagged ensemble. Only bagged models (such a stacker models) can produce OOF predictions.
+            This also means that refit_full models and distilled models will raise an exception.
+        Warning: If intending to join the output of this method with the original training data, be aware that a rare edge-case issue exists:
+            Multiclass problems with rare classes combined with the use of the 'log_loss' eval_metric may have forced AutoGluon to duplicate rows in the training data to satisfy minimum class counts in the data.
+            If this has occurred, then the indices and row counts of the returned pandas Series in this method may not align with the training data.
+            In this case, consider fetching the processed training data using `predictor.load_data_internal()` instead of using the original training data.
+            A more benign version of this issue occurs when 'log_loss' wasn't specified as the eval_metric but rare classes were dropped by AutoGluon.
+            In this case, not all of the original training data rows will have an OOF prediction. It is recommended to either drop these rows during the join or to get direct predictions on the missing rows via `predictor.predict`.
+
+        Parameters
+        ----------
+        model : str (optional)
+            The name of the model to get out-of-fold predictions from. Defaults to None, which uses the highest scoring model on the validation set.
+            Valid models are listed in this `predictor` by calling `predictor.get_model_names()`
+        transformed : bool, default = False
+            Whether the output values should be of the external label representation (False) or the internal label representation (True).
```

Comment:
Clarify external label representation = original class names, internal label representation = integers numbering the possible classes from 1,...,K

Reply:
Added